resolwe/elastic/signals.py

```diff
@@ -0,0 +1,76 @@
+""".. Ignore pydocstyle D400.
+
+===============
+Signal Handlers
+===============
+
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from django.db.models.signals import pre_delete, post_save
+from django.dispatch import receiver
+
+from guardian.models import GroupObjectPermission, UserObjectPermission
+
+from resolwe.flow.models import Data, Sample
+
+from .utils import add_index, delete_index
+
+
+def _process_permission(perm):
+    if not perm.permission.codename.startswith('view'):
+        return
+
+    if not perm.content_type.name in ['data', 'sample']:
+        return
+
+    obj = perm.content_type.get_object_for_this_type(id=perm.object_pk)
+    add_index(obj)
+
+
+@receiver(post_save, sender=Data)
```

Comment:
I don't like these hardcoded signals. Can we make it so that the first time a new index references a new model, the appropriate signals are connected?